### ISSUE TITLE
Implement clone-censor-weight methods

### ISSUE THREAD

**Issue by gravesti** [Causal-LDA/TrialEmulation#115]
- [ ] add reference
- [ ] See https://github.com/lpetito/SEERMedicareCEAnalysis/blob/master/10_perprotocol.R

**Comment by lisu-stats** [comment]
@gravesti I have tried to simulate some data to mimic the scenario that patients' data are compatible with multiple dynamic treatment regimes.   For example, the columns `compatible`, `compatible1` below represent whether patients data are consistent with two regimes based on two time-varying covariates X1 (binary), X2(continuous), 

1. Intiate treatment at the first visit with X2 >0
2. Intiate treatment at the first visit with X2 >0 or X1=1

![image](https://github.com/user-attachments/assets/aae89817-c798-42e3-9bb8-d1701f2ca3a6)


However, we need to calculate weights for the observed treatments up to the last visit that the regime is  adhered. This is not the same as we do in the package because we only calculate for the **always treated** vs **always controlled** regimes.  So we will need to alter the weight calculation when applying clone-censor-weight because of the difference in how data are coded for treatments and a particular regime. I will work out an example using some generic coding in R.

**Comment by gravesti** [comment]
@lisu-stats I guess from this data set up, we would just need to stack the data, once with `compatible` and then with `compatible1`, and modify the IDs for the 2nd copy. Then I think it could work directly in our existing functions.

Maybe we would need to change the IDs back for the sandwich estimator, so the clones have the same ID?

**Comment by lisu-stats** [comment]
@gravesti In this simulation setting, the true weights should be the IPTW weights. Simply fitting a IPCW model to `compatible` and `compatible1` may have the risk of model misspecification. Note that in these simulated data, there was more variation in the treatment process.  See the comment from Gaber et al's review paper

![image](https://github.com/user-attachments/assets/8b8abf56-654a-4698-9bf6-be7dcc1e5b6b)

I will discuss this issue with Shaun.

**Comment by lisu-stats** [comment]
@gravesti Shaun and I agreed that we probably want to implement IPTW and IPCW for the clone-censor-weight approach. 

IPCW is straightforward： as you mentioned, we just calculate the weights separately based on created censoring indicators `compatible` and `compatible1`. Then weights are applied to the cloned data with an MSM. Then we use the fitted MSM to predict survival curves.  You are quite right, clones should have the same IDs as they come from the same patients. 

For IPTW, we need to use the original treatment data before cloning and artificial censoring to estimate the treatment probability. For each uncensored observation in cloned data,  we use these treatment models to  estimate the probability of being uncensored up to visit `t` according to the observed treatment sequence up to `t`. 

Because of the inference problems for weights estimated by data-adaptive methods as mentioned above, let's focus on parametric models for weight estimation.  Adapting undersmoothed HAL to longitudinal settings will be a big step and we will see if there is new work on this coming out.

**Comment by gravesti** [comment]
@lisu-stats Thanks for the explanation. I think I will need to talk through the IPTW next week.

I have also see this pre-print manuscript https://arxiv.org/pdf/2404.15073 where they suggest to limit the patients who are included in the IPCW calculation to avoid an "impossible intervention". I'd be pleased to hear what you think about this.

**Comment by lisu-stats** [comment]
@gravesti I agree it is tricky when the treatment timing is incorporated in a regimen. For such regimens, I feel like alternative approach such as longitudinal modified treatment policy might be better https://muse.jhu.edu/article/883479

**Comment by lisu-stats** [comment]
@gravesti In IPCW, we will need to correct the selection bias by censoring those who did not initiate treatment at the end of a grace period in terms of all observed histories up to the censoring time. This will include treatment exposure history, baseline covariates and time-varying covariates. By doing so, we will only use the outcome data from those who were most similar to the censored patients to predict the outcome data after the grace period, assuming that they initiate treatments immediately at the end of the grace period.
